File: Week1/Codes/talker.py
#!/usr/bin/env python3

import rospy
from code.msg import MyCustomMsg

def publisher():
    rospy.init_node('publisher', anonymous=True)
    pub = rospy.Publisher('custom_topic', MyCustomMsg, queue_size=10)
    rate = rospy.Rate(1) 

    while not rospy.is_shutdown():
        msg = MyCustomMsg()
        msg.my_int = 1632
        msg.my_str = "It works,thank god"
        msg.my_bool = True
        rospy.loginfo("Publishing: %s", msg)
        pub.publish(msg)
        rate.sleep()

if __name__ == '__main__':
    try:
        publisher()
    except rospy.ROSInterruptException:
        pass

File: Week1/Codes/talker.cpp
// talker.cpp
#include "ros/ros.h"
#include "code/MyCustomMsg.h"

int main(int argc, char **argv)
{


  ros::init(argc, argv, "talker");
  ros::NodeHandle n;
  ros::Publisher cuspub=n.advertise<code::MyCustomMsg>("custom_topic", 10);

  ros::Rate loop_rate(10);



  while (ros::ok())
  {
    code::MyCustomMsg msg;
    msg.my_int=2402;
    msg.my_str="It finally works";
    msg.my_bool=true;




    ROS_INFO("Publishing: %d, %s, %s",msg.my_int,msg.my_str.c_str(),msg.my_bool ? "true" : "false");
    cuspub.publish(msg);
    ros::spinOnce();


    loop_rate.sleep();
  }

  return 0;
}
